event.go

```diff
@@ -46,6 +49,32 @@ func NewEvent(t EventType, o Identifiable) *Event {
 	return NewEventWithEncoding(t, o, EncodingTypeMSGPACK)
 }
 
+// NewErrorEvent returns a new (error) Event embedded with the provided elemental.Error
+func NewErrorEvent(ee Error, encoding EncodingType) *Event {
+
+	data, err := Encode(encoding, ee)
+	if err != nil {
+		panic(fmt.Sprintf("unable to create new error event: %s", err))
```

Comment:
@primalmotion code coverage check is failing because of this line as I can't find a way to make [the call to Encode](https://github.com/aporeto-inc/elemental/pull/65/files#diff-9fb85d1ee9f1b0a59804bec41e1491feR55) return an error.

Reply:
I believe you can do it by making an Error that implemented the MarshalJSON interface that returns an error

Reply:
Mmm, currently this function accepts an elemental error as an argument, in order to do what you are suggesting, I would have to change it so it takes an error (the interface).

That may make the API too loose as we want to force the usage of an elemental error.

What do you think?

Reply:
that's correct. let's leave it as is